### ISSUE TITLE
(Passive) Mobs Don't Move In Space.

### ISSUE THREAD

**Issue by BumblebeeCannon** [micdoodle8/Galacticraft#2069]
So I was setting up a sheep farm on my space station, and after a while, I realized that none of the sheep were walking at all, even when they are planted on the ground. I did some testing, this same thing happens with horses, cows, chickens, etc, however, I didn't get to test this with any hostile mobs, yet. These mobs can still turn, but their pathfinding won't let them move anywhere. Hope to see this fixed soon, thanks!

**Comment by EzerArch** [comment]
The passive mobs didn't show any animation at all or just they didn't move?

**Comment by BumblebeeCannon** [comment]
They still show animations, but not walking animations, they just can't move.

**Comment by EzerArch** [comment]
It's reasonable that mobs can't move in zero gravity (they can in non-zero gravity dimensions). No friction = no mobility, players can "grab" the structure around to move. But the mechanics is a bit inconsistent.

Those entities can't move, but the living animation still works:
* All hostile mobs, except at least spiders, endermen and silverfish.
* Animals, they can spot their food (wheat, seeds, carrots), but can't move at all. You can push and leash them.
* Passive mobs, except bats.

Meanwhile you can leash the animals or install gravity gun mod to move any mob.

**Comment by BumblebeeCannon** [comment]
Well, that isn't exactly true. The normal reaction force is not necessarily equal to weight. When you jump, you push down on the ground. That pushing force plus their weight result in a normal reaction force larger than your weight, which is why you are propelled off the ground, therefore, they would still be able to somewhat move, or at least float. They wouldn't just sit on the ground without moving. So I don't really think this is an intended feature.

**Comment by radfast** [comment]
>  or at least float.

Be careful what you wish for ...

**Comment by EzerArch** [comment]
When I said "No friction = no mobility", I didn't include jumping (vertical movement) because the issue is "mobs can't walk", walk as in "movement parallel to the floor".

> The normal reaction force is not necessarily equal to weight. When you jump, you push down on the ground.

I know that, the problem is: mobs, except a few ones like spiders and silverfish, don't jump (that is, "push down on the ground"). Thank goodness they can't, otherwise you'd your cattle stuck on the ceiling after they attempted to jump. Unless they learn to flip upside down and jump downwards. So...

> Be careful what you wish for...

:)

**Comment by BumblebeeCannon** [comment]
True, maybe they could have the same type of gravity as item entities? The gravity could be reduced, but will still fall downwards, relevant to Earth. That could help solve this issue.

**Comment by XFactHD** [comment]
I wouldn't say that a space station is a zero gravity dimension, because everything that has mass creates gravitational forces.

**Comment by EzerArch** [comment]
If I had to choose, I would prefer that any mob could walk as players can: if on floor and with sluggish movement, but not drifting nor floating randomly (they could float if something makes them). I don't want a random pig flying on my face. O.O'

Items could float instead of falling. [That'll be](https://www.youtube.com/watch?v=W1lkeM6YoqU) [awesome](https://www.youtube.com/watch?v=o8TssbmY-GM). I hate having them falling back to the OW.

**Comment by C1ffisme** [comment]
Erm... Are you fixing this bug based on the code, or realism?

If the player can move in space, then it seems that the animals should also be able to.

Heck, if the mod is going to be realistic, please make all rockets 50x bigger, remove aliens, slimelings, dungeons, etc. and make the player have to worry about the heat and pressure of the rocket. And the politicians governing him.

**Comment by EzerArch** [comment]
@C1ffisme, this mods is not going to be 100% realistic. We pick real elements when they fit. In game design, we use science as an excuse, not as a ruler.

**Comment by dj3520** [comment]
Is the testing done inside a sealed space? I've yet to do a test I've wanted to for a while now to see if any mobs will pathfind through sealed air since it's technically a block. I believe I tested this a while ago and the result was no, but my memory of anything else regarding the test (if it was 1.6.4 or 1.7.10 even) is gone.

>I know that, the problem is: mobs, except a few ones like spiders and silverfish, don't jump

You might be forgetting that mobs will jump to try to go up 1 block high steps.

>If the player can move in space, then it seems that the animals should also be able to.

I agree, but the reason a player can move in space when not touching the ground is "imaginary thrusters" that are included to make it so the experience isn't super annoying. IMO mobs touching the floor should be able to move normally, including the up/down one block jump, if not just to make things less of a headache. However if they are forced upwards somehow (player hit, gravity gun, explosion, etc) and become disconnected with the floor, the current behavior of helplessly floating should stay. Leashing or pushing them back down to the floor would fix this. This is somewhat like how chickens currently work, except that also unsealing the space causes them to fall back down. Not completely sure about the science behind that but I'm guessing it's suffocation damage. Technically chickens should float upwards indefinitely when in a sealed space. Discussion and reasoning about this can be found in #1614

**Comment by BumblebeeCannon** [comment]
@dj3520 Yes, this testubg was done in a sealed environment.

**Comment by C1ffisme** [comment]
@EzerArch Okay, it just seems that not fixing something like this because of science is really not a good reason.

I didn't mean that we were actually trying to be realistic, just that if we were, then those things would happen.

**Comment by dj3520** [comment]
@C1ffisme I think it's implied by their existence in the first place that realism is the goal in certain aspects and not at all in others. While realism is going to play a part, a game is still a game, so it might as well have a few quirks in it. Plus, minecraft itself isn't realistic.

**Comment by EzerArch** [comment]
I suggest closing this for being too old and as "intended behavior", at least for 1.7.10.

Workaround if you want to move animals in zero-G: use leash or Gravity Gun mod (won't work if the animal is from other dimension, due to NBTdata dimID).
